tests/usbus/usbdev_mock.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2019 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @defgroup    tests_usbdev_mock USBdev mockup device
+ * @ingroup     tests
+ * @brief       USBdev mockup device for testing
+ * @{
+ *
+ * @file
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+
+#ifndef USBDEV_MOCK_H
+#define USBDEV_MOCK_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Number of IN and OUT endpoints available in the mock usbdev device
+ */
+#define USBDEV_MOCK_NUM_EP      (1)
+
+/**
+ * @name usbdev mock device endpoint states
+ */
+typedef enum {
+    EP_STATE_READY,             /**< Endpoint is ready/idle */
+    EP_STATE_DATA_AVAILABLE,    /**< Endpoint has data      */
+} usbdev_mock_ep_state_t;
+
+/**
+ * @brief usbdev mock device endpoint
+ */
+typedef struct {
+    usbdev_ep_t ep;                 /**< Generic endpoint struct        */
+    usbdev_mock_ep_state_t state;   /**< Endpoint state                 */
+    size_t available;               /**< Bytes available in the buffer  */
+    uint8_t *buf_start;             /**< Start location of the buffer   */
+} usbdev_mock_ep_t;
+
+/**
+ * @brief usbdev mock device forward declaration
+ */
+typedef struct usbdev_mock usbdev_mock_t;
+
+/**
+ * @brief usbdev mock device callback for esr event checking
+ *
+ * @param dev   usbdev mock device state
```

Comment:
You should change this description
Same for all functions within this file using dev arg

Reply:
Modified